api/index.js

```diff
@@ -12,24 +12,73 @@ const ews = require('ews-javascript-api');
 const azureStorage = require('azure-storage');
 const sha256 = require('sha256');
 const intoStream = require('into-stream');
+const jsonwebtoken = require('jsonwebtoken');
+const jsrsasign = require('jsrsasign');
+const url = require('url');
+const uuidv4 = require('uuid/v4');
 
 var utils = require('../utils');
-
 var app = express();
 
 const iberaServicesEndpoint = config.IBERA_SERVICES_ENDPOINT;
 const azureStorageConnectionString = config.STORAGE_CONNECTION_STRING;
 
-const USER_TOKEN_HEADER_KEY = 'user-token';
```

Comment:
Since there's relatively a lot of code dealing with authentication, I would move it to a separate file

Reply:
done!
Now the authorization model is done in auth.js